zerver/actions/message_delete.py

```diff
@@ -146,3 +147,33 @@ def do_delete_messages_by_sender(user: UserProfile) -> None:
     )
     if message_ids:
         move_messages_to_archive(message_ids, chunk_size=retention.STREAM_MESSAGE_BATCH_SIZE)
+
+
+def delete_deactivated_user_messages(
+    realm: Realm, user_profile: UserProfile, message_delete_action: list[int]
+) -> None:
+    if Message.NO_DELETE_ACTION in message_delete_action:
+        return
+    user_stream_messages = list(
+        Message.objects.filter(
+            realm_id=user_profile.realm_id,
+            sender=user_profile,
+            recipient__type=Recipient.STREAM,
+        ).only("id", "recipient__type", "recipient__type_id")
+    )
+    private_stream_messages, public_stream_messages = classify_stream_messages(
+        realm, user_stream_messages
+    )
+    if Message.DELETE_PUBLIC_STREAM_MESSAGES in message_delete_action:
+        do_delete_messages(user_profile.realm, public_stream_messages, acting_user=user_profile)
+    if Message.DELETE_PRIVATE_STREAM_MESSAGES in message_delete_action:
+        do_delete_messages(user_profile.realm, private_stream_messages, acting_user=user_profile)
```

Comment:
I guess `do_delete_messages` can be called just once atleast for private and public stream messages since `do_delete_messages` handles the groupings. Maybe you can part streams and group DMs together to `do_delete_messages` and 1:1 DMs separately by grouping as per the other user in 1:1 DM.

Reply:
Now `do_delete_messages` fetches each stream in a different query when messages from multiple streams are passed together, so maybe we can pass it an optional parameter in the form of a `dict[int, Stream]` so that it can be used directly.